src/Control/Carrier/Writer/Strict.hs

```diff
@@ -60,15 +61,18 @@ newtype WriterC w m a = WriterC { runWriterC :: StateC w m a }
   deriving (Alternative, Applicative, Functor, Monad, Fail.MonadFail, MonadFix, MonadIO, MonadPlus, MonadTrans)
 
 instance (Monoid w, Algebra sig m) => Algebra (Writer w :+: sig) (WriterC w m) where
-  alg hdl sig ctx = case sig of
-    L (Tell w)     -> ctx <$ WriterC (modify (`mappend` w))
-    L (Listen   m) -> WriterC . StateC $ \ w -> do
-      (w', a) <- runWriter (hdl (m <$ ctx))
-      let w'' = mappend w w'
-      w'' `seq` pure (w'', (,) w' <$> a)
-    L (Censor f m) -> WriterC . StateC $ \ w -> do
-      (w', a) <- runWriter (hdl (m <$ ctx))
-      let w'' = mappend w (f w')
-      w'' `seq` pure (w'', a)
-    R other        -> WriterC (alg (runWriterC . hdl) (R other) ctx)
+  alg hdl sig ctx = WriterC $ case sig of
+    L writer -> StateC $ \ w -> case writer of
+      Tell w'    -> do
+        let !w'' = mappend w w'
+        pure (w'', ctx)
```

Comment:
`mappend`ing strictly here made for a ~2.5x improvement to the performance of this carrier in the benchmarks.